src/mrinufft/operators/interfaces/cufinufft.py

```diff
@@ -89,12 +90,31 @@ def _make_plan(self, typ, **kwargs):
             **kwargs,
         )
 
-    def _set_pts(self, typ):
-        self.plans[typ].setpts(
-            cp.array(self.samples[:, 0], copy=False),
-            cp.array(self.samples[:, 1], copy=False),
-            cp.array(self.samples[:, 2], copy=False) if self.ndim == 3 else None,
+    def _make_plan_grad(self, **kwargs):
+        self.grad_plan = Plan(
+            2,
+            self.shape,
+            self.n_trans,
+            self.eps,
+            dtype=DTYPE_R2C[str(self.samples.dtype)],
+            isign=1,
+            **kwargs,
         )
+        self._set_pts(typ="grad")
+
+    def _set_pts(self, typ):
+        if typ == "grad":
+            self.grad_plan.setpts(
+                cp.array(self.samples[:, 0], copy=False),
+                cp.array(self.samples[:, 1], copy=False),
+                cp.array(self.samples[:, 2], copy=False) if self.ndim == 3 else None,
+            )
+        else:
+            self.plans[typ].setpts(
+                cp.array(self.samples[:, 0], copy=False),
+                cp.array(self.samples[:, 1], copy=False),
+                cp.array(self.samples[:, 2], copy=False) if self.ndim == 3 else None,
+            )
 
     def _destroy_plan(self, typ):
         if self.plans[typ] is not None:
```

Comment:
Perhaps, you might need to also need to write a way to destroy grad_plan

Reply:
> Perhaps, you might need to also need to write a way to destroy grad_plan

I have added this function in cufinufft, but I am not sure if I need to call this function in any other files...